File: mdm/src/masterData.service.js
const store = new Map();

export default {
  async create(data) {
    store.set(data.code, data);
    return data;
  },

  async findAll() {
    return Array.from(store.values());
  },

  async findOne(code) {
    return store.get(code);
  },

  async update(code, data) {
    if (!store.has(code)) return null;
    store.set(code, data);
    return data;
  },

  async remove(code) {
    if (!store.has(code)) return null;
    store.delete(code);
    return true;
  }
};

File: mdm/src/main/java/com/mdm/mdm/service/MasterDataService.java
package com.mdm.mdm.service;

import com.mdm.mdm.repository.MasterDataRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.Optional;
import com.mdm.mdm.dto.MasterDataDTO;

@Service
public class MasterDataService {

    private final MasterDataRepository masterDataRepository;

    public MasterDataService(MasterDataRepository masterDataRepository) {
        this.masterDataRepository = masterDataRepository;
    }

    public MasterDataDTO create(MasterDataDTO data) {
        data.setCreatedAt(LocalDateTime.now());
        data.setUpdatedAt(LocalDateTime.now());
        return masterDataRepository.save(data);
    }

    public Iterable<MasterDataDTO> findAll() {
        return masterDataRepository.findAll();
    }

    public Optional<MasterDataDTO> findOne(String code) {
        return masterDataRepository.findById(code);
    }

    public Optional<MasterDataDTO> update(String code, MasterDataDTO data) {
        if (!masterDataRepository.existsById(code)) {
            return Optional.empty();
        }
        data.setCountryID(code);
        data.setCreatedAt(masterDataRepository.findById(code).get().getCreatedAt());
        data.setUpdatedAt(LocalDateTime.now());
        return Optional.of(masterDataRepository.save(data));
    }

    public boolean delete(String code) {
        if (masterDataRepository.existsById(code)) {
            masterDataRepository.deleteById(code);
            return true;
        }
        return false;
    }
}
